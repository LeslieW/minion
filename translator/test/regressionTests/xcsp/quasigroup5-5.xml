<?xml version="1.0" encoding="UTF-8"?>

<instance>
<presentation maxConstraintArity="6" format="XCSP 2.1"/>

  <domains nbDomains="16">
    <domain name="D0" nbValues="1">0</domain>
    <domain name="D1" nbValues="3">2..4</domain>
    <domain name="D2" nbValues="3">1 3 4</domain>
    <domain name="D3" nbValues="3">1 2 4</domain>
    <domain name="D4" nbValues="3">1..3</domain>
    <domain name="D5" nbValues="1">1</domain>
    <domain name="D6" nbValues="3">0 3 4</domain>
    <domain name="D7" nbValues="3">0 2 4</domain>
    <domain name="D8" nbValues="3">0 2 3</domain>
    <domain name="D9" nbValues="1">2</domain>
    <domain name="D10" nbValues="3">0 1 4</domain>
    <domain name="D11" nbValues="3">0 1 3</domain>
    <domain name="D12" nbValues="1">3</domain>
    <domain name="D13" nbValues="3">0..2</domain>
    <domain name="D14" nbValues="1">4</domain>
    <domain name="D15" nbValues="5">0..4</domain>
  </domains>

  <variables nbVariables="65">
    <variable name="V0" domain="D0"/>
    <variable name="V1" domain="D1"/>
    <variable name="V2" domain="D2"/>
    <variable name="V3" domain="D3"/>
    <variable name="V4" domain="D4"/>
    <variable name="V5" domain="D1"/>
    <variable name="V6" domain="D5"/>
    <variable name="V7" domain="D6"/>
    <variable name="V8" domain="D7"/>
    <variable name="V9" domain="D8"/>
    <variable name="V10" domain="D2"/>
    <variable name="V11" domain="D6"/>
    <variable name="V12" domain="D9"/>
    <variable name="V13" domain="D10"/>
    <variable name="V14" domain="D11"/>
    <variable name="V15" domain="D3"/>
    <variable name="V16" domain="D7"/>
    <variable name="V17" domain="D10"/>
    <variable name="V18" domain="D12"/>
    <variable name="V19" domain="D13"/>
    <variable name="V20" domain="D4"/>
    <variable name="V21" domain="D8"/>
    <variable name="V22" domain="D11"/>
    <variable name="V23" domain="D13"/>
    <variable name="V24" domain="D14"/>
    <variable name="V25" domain="D15"/>
    <variable name="V26" domain="D5"/>
    <variable name="V27" domain="D15"/>
    <variable name="V28" domain="D9"/>
    <variable name="V29" domain="D15"/>
    <variable name="V30" domain="D12"/>
    <variable name="V31" domain="D15"/>
    <variable name="V32" domain="D14"/>
    <variable name="V33" domain="D15"/>
    <variable name="V34" domain="D0"/>
    <variable name="V35" domain="D15"/>
    <variable name="V36" domain="D9"/>
    <variable name="V37" domain="D15"/>
    <variable name="V38" domain="D12"/>
    <variable name="V39" domain="D15"/>
    <variable name="V40" domain="D14"/>
    <variable name="V41" domain="D15"/>
    <variable name="V42" domain="D0"/>
    <variable name="V43" domain="D15"/>
    <variable name="V44" domain="D5"/>
    <variable name="V45" domain="D15"/>
    <variable name="V46" domain="D12"/>
    <variable name="V47" domain="D15"/>
    <variable name="V48" domain="D14"/>
    <variable name="V49" domain="D15"/>
    <variable name="V50" domain="D0"/>
    <variable name="V51" domain="D15"/>
    <variable name="V52" domain="D5"/>
    <variable name="V53" domain="D15"/>
    <variable name="V54" domain="D9"/>
    <variable name="V55" domain="D15"/>
    <variable name="V56" domain="D14"/>
    <variable name="V57" domain="D15"/>
    <variable name="V58" domain="D0"/>
    <variable name="V59" domain="D15"/>
    <variable name="V60" domain="D5"/>
    <variable name="V61" domain="D15"/>
    <variable name="V62" domain="D9"/>
    <variable name="V63" domain="D15"/>
    <variable name="V64" domain="D12"/>
  </variables>

  <constraints nbConstraints="50">
    <constraint name="C0" arity="7" scope="V20 V21 V22 V23 V24 V63 V64 " reference="global:element">
      <parameters>
        V63
        [ V20 V21 V22 V23 V24 ]
        V64
      </parameters>
    </constraint>
    <constraint name="C1" arity="7" scope="V19 V20 V21 V22 V23 V24 V63 " reference="global:element">
      <parameters>
        V19
        [ V20 V21 V22 V23 V24 ]
        V63
      </parameters>
    </constraint>
    <constraint name="C2" arity="7" scope="V20 V21 V22 V23 V24 V61 V62 " reference="global:element">
      <parameters>
        V61
        [ V20 V21 V22 V23 V24 ]
        V62
      </parameters>
    </constraint>
    <constraint name="C3" arity="7" scope="V14 V20 V21 V22 V23 V24 V61 " reference="global:element">
      <parameters>
        V14
        [ V20 V21 V22 V23 V24 ]
        V61
      </parameters>
    </constraint>
    <constraint name="C4" arity="7" scope="V20 V21 V22 V23 V24 V59 V60 " reference="global:element">
      <parameters>
        V59
        [ V20 V21 V22 V23 V24 ]
        V60
      </parameters>
    </constraint>
    <constraint name="C5" arity="7" scope="V9 V20 V21 V22 V23 V24 V59 " reference="global:element">
      <parameters>
        V9
        [ V20 V21 V22 V23 V24 ]
        V59
      </parameters>
    </constraint>
    <constraint name="C6" arity="7" scope="V20 V21 V22 V23 V24 V57 V58 " reference="global:element">
      <parameters>
        V57
        [ V20 V21 V22 V23 V24 ]
        V58
      </parameters>
    </constraint>
    <constraint name="C7" arity="7" scope="V4 V20 V21 V22 V23 V24 V57 " reference="global:element">
      <parameters>
        V4
        [ V20 V21 V22 V23 V24 ]
        V57
      </parameters>
    </constraint>
    <constraint name="C8" arity="7" scope="V15 V16 V17 V18 V19 V55 V56 " reference="global:element">
      <parameters>
        V55
        [ V15 V16 V17 V18 V19 ]
        V56
      </parameters>
    </constraint>
    <constraint name="C9" arity="7" scope="V15 V16 V17 V18 V19 V23 V55 " reference="global:element">
      <parameters>
        V23
        [ V15 V16 V17 V18 V19 ]
        V55
      </parameters>
    </constraint>
    <constraint name="C10" arity="7" scope="V15 V16 V17 V18 V19 V53 V54 " reference="global:element">
      <parameters>
        V53
        [ V15 V16 V17 V18 V19 ]
        V54
      </parameters>
    </constraint>
    <constraint name="C11" arity="7" scope="V13 V15 V16 V17 V18 V19 V53 " reference="global:element">
      <parameters>
        V13
        [ V15 V16 V17 V18 V19 ]
        V53
      </parameters>
    </constraint>
    <constraint name="C12" arity="7" scope="V15 V16 V17 V18 V19 V51 V52 " reference="global:element">
      <parameters>
        V51
        [ V15 V16 V17 V18 V19 ]
        V52
      </parameters>
    </constraint>
    <constraint name="C13" arity="7" scope="V8 V15 V16 V17 V18 V19 V51 " reference="global:element">
      <parameters>
        V8
        [ V15 V16 V17 V18 V19 ]
        V51
      </parameters>
    </constraint>
    <constraint name="C14" arity="7" scope="V15 V16 V17 V18 V19 V49 V50 " reference="global:element">
      <parameters>
        V49
        [ V15 V16 V17 V18 V19 ]
        V50
      </parameters>
    </constraint>
    <constraint name="C15" arity="7" scope="V3 V15 V16 V17 V18 V19 V49 " reference="global:element">
      <parameters>
        V3
        [ V15 V16 V17 V18 V19 ]
        V49
      </parameters>
    </constraint>
    <constraint name="C16" arity="7" scope="V10 V11 V12 V13 V14 V47 V48 " reference="global:element">
      <parameters>
        V47
        [ V10 V11 V12 V13 V14 ]
        V48
      </parameters>
    </constraint>
    <constraint name="C17" arity="7" scope="V10 V11 V12 V13 V14 V22 V47 " reference="global:element">
      <parameters>
        V22
        [ V10 V11 V12 V13 V14 ]
        V47
      </parameters>
    </constraint>
    <constraint name="C18" arity="7" scope="V10 V11 V12 V13 V14 V45 V46 " reference="global:element">
      <parameters>
        V45
        [ V10 V11 V12 V13 V14 ]
        V46
      </parameters>
    </constraint>
    <constraint name="C19" arity="7" scope="V10 V11 V12 V13 V14 V17 V45 " reference="global:element">
      <parameters>
        V17
        [ V10 V11 V12 V13 V14 ]
        V45
      </parameters>
    </constraint>
    <constraint name="C20" arity="7" scope="V10 V11 V12 V13 V14 V43 V44 " reference="global:element">
      <parameters>
        V43
        [ V10 V11 V12 V13 V14 ]
        V44
      </parameters>
    </constraint>
    <constraint name="C21" arity="7" scope="V7 V10 V11 V12 V13 V14 V43 " reference="global:element">
      <parameters>
        V7
        [ V10 V11 V12 V13 V14 ]
        V43
      </parameters>
    </constraint>
    <constraint name="C22" arity="7" scope="V10 V11 V12 V13 V14 V41 V42 " reference="global:element">
      <parameters>
        V41
        [ V10 V11 V12 V13 V14 ]
        V42
      </parameters>
    </constraint>
    <constraint name="C23" arity="7" scope="V2 V10 V11 V12 V13 V14 V41 " reference="global:element">
      <parameters>
        V2
        [ V10 V11 V12 V13 V14 ]
        V41
      </parameters>
    </constraint>
    <constraint name="C24" arity="7" scope="V5 V6 V7 V8 V9 V39 V40 " reference="global:element">
      <parameters>
        V39
        [ V5 V6 V7 V8 V9 ]
        V40
      </parameters>
    </constraint>
    <constraint name="C25" arity="7" scope="V5 V6 V7 V8 V9 V21 V39 " reference="global:element">
      <parameters>
        V21
        [ V5 V6 V7 V8 V9 ]
        V39
      </parameters>
    </constraint>
    <constraint name="C26" arity="7" scope="V5 V6 V7 V8 V9 V37 V38 " reference="global:element">
      <parameters>
        V37
        [ V5 V6 V7 V8 V9 ]
        V38
      </parameters>
    </constraint>
    <constraint name="C27" arity="7" scope="V5 V6 V7 V8 V9 V16 V37 " reference="global:element">
      <parameters>
        V16
        [ V5 V6 V7 V8 V9 ]
        V37
      </parameters>
    </constraint>
    <constraint name="C28" arity="7" scope="V5 V6 V7 V8 V9 V35 V36 " reference="global:element">
      <parameters>
        V35
        [ V5 V6 V7 V8 V9 ]
        V36
      </parameters>
    </constraint>
    <constraint name="C29" arity="7" scope="V5 V6 V7 V8 V9 V11 V35 " reference="global:element">
      <parameters>
        V11
        [ V5 V6 V7 V8 V9 ]
        V35
      </parameters>
    </constraint>
    <constraint name="C30" arity="7" scope="V5 V6 V7 V8 V9 V33 V34 " reference="global:element">
      <parameters>
        V33
        [ V5 V6 V7 V8 V9 ]
        V34
      </parameters>
    </constraint>
    <constraint name="C31" arity="7" scope="V1 V5 V6 V7 V8 V9 V33 " reference="global:element">
      <parameters>
        V1
        [ V5 V6 V7 V8 V9 ]
        V33
      </parameters>
    </constraint>
    <constraint name="C32" arity="7" scope="V0 V1 V2 V3 V4 V31 V32 " reference="global:element">
      <parameters>
        V31
        [ V0 V1 V2 V3 V4 ]
        V32
      </parameters>
    </constraint>
    <constraint name="C33" arity="7" scope="V0 V1 V2 V3 V4 V20 V31 " reference="global:element">
      <parameters>
        V20
        [ V0 V1 V2 V3 V4 ]
        V31
      </parameters>
    </constraint>
    <constraint name="C34" arity="7" scope="V0 V1 V2 V3 V4 V29 V30 " reference="global:element">
      <parameters>
        V29
        [ V0 V1 V2 V3 V4 ]
        V30
      </parameters>
    </constraint>
    <constraint name="C35" arity="7" scope="V0 V1 V2 V3 V4 V15 V29 " reference="global:element">
      <parameters>
        V15
        [ V0 V1 V2 V3 V4 ]
        V29
      </parameters>
    </constraint>
    <constraint name="C36" arity="7" scope="V0 V1 V2 V3 V4 V27 V28 " reference="global:element">
      <parameters>
        V27
        [ V0 V1 V2 V3 V4 ]
        V28
      </parameters>
    </constraint>
    <constraint name="C37" arity="7" scope="V0 V1 V2 V3 V4 V10 V27 " reference="global:element">
      <parameters>
        V10
        [ V0 V1 V2 V3 V4 ]
        V27
      </parameters>
    </constraint>
    <constraint name="C38" arity="7" scope="V0 V1 V2 V3 V4 V25 V26 " reference="global:element">
      <parameters>
        V25
        [ V0 V1 V2 V3 V4 ]
        V26
      </parameters>
    </constraint>
    <constraint name="C39" arity="7" scope="V0 V1 V2 V3 V4 V5 V25 " reference="global:element">
      <parameters>
        V5
        [ V0 V1 V2 V3 V4 ]
        V25
      </parameters>
    </constraint>
    <constraint name="C40" arity="4" scope="V4 V9 V14 V19 " reference="global:allDifferent"/>
    <constraint name="C41" arity="4" scope="V20 V21 V22 V23 " reference="global:allDifferent"/>
    <constraint name="C42" arity="4" scope="V3 V8 V13 V23 " reference="global:allDifferent"/>
    <constraint name="C43" arity="4" scope="V15 V16 V17 V19 " reference="global:allDifferent"/>
    <constraint name="C44" arity="4" scope="V2 V7 V17 V22 " reference="global:allDifferent"/>
    <constraint name="C45" arity="4" scope="V10 V11 V13 V14 " reference="global:allDifferent"/>
    <constraint name="C46" arity="4" scope="V1 V11 V16 V21 " reference="global:allDifferent"/>
    <constraint name="C47" arity="4" scope="V5 V7 V8 V9 " reference="global:allDifferent"/>
    <constraint name="C48" arity="4" scope="V5 V10 V15 V20 " reference="global:allDifferent"/>
    <constraint name="C49" arity="4" scope="V1 V2 V3 V4 " reference="global:allDifferent"/>
  </constraints>

</instance>
