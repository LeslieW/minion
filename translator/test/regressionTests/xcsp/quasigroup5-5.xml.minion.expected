MINION 3

#This file has been automatically generated by TAILOR v0.2
#bug-reports: andrea@cs.st-and.ac.uk

# amount of common subexpressions used:0
# amount of inferred common subexpressions used:0
# amount of original variables saved (over direct equality):0
# amount of constraints:50
# amount of auxiliary variables: 0

**VARIABLES**
DISCRETE V0 {0..0}
DISCRETE V1 {2..4}
DISCRETE V2 {1..4}
DISCRETE V3 {1..4}
DISCRETE V4 {1..3}
DISCRETE V5 {2..4}
DISCRETE V6 {1..1}
DISCRETE V7 {0..4}
DISCRETE V8 {0..4}
DISCRETE V9 {0..3}
DISCRETE V10 {1..4}
DISCRETE V11 {0..4}
DISCRETE V12 {2..2}
DISCRETE V13 {0..4}
DISCRETE V14 {0..3}
DISCRETE V15 {1..4}
DISCRETE V16 {0..4}
DISCRETE V17 {0..4}
DISCRETE V18 {3..3}
DISCRETE V19 {0..2}
DISCRETE V20 {1..3}
DISCRETE V21 {0..3}
DISCRETE V22 {0..3}
DISCRETE V23 {0..2}
DISCRETE V24 {4..4}
DISCRETE V25 {0..4}
DISCRETE V26 {1..1}
DISCRETE V27 {0..4}
DISCRETE V28 {2..2}
DISCRETE V29 {0..4}
DISCRETE V30 {3..3}
DISCRETE V31 {0..4}
DISCRETE V32 {4..4}
DISCRETE V33 {0..4}
DISCRETE V34 {0..0}
DISCRETE V35 {0..4}
DISCRETE V36 {2..2}
DISCRETE V37 {0..4}
DISCRETE V38 {3..3}
DISCRETE V39 {0..4}
DISCRETE V40 {4..4}
DISCRETE V41 {0..4}
DISCRETE V42 {0..0}
DISCRETE V43 {0..4}
DISCRETE V44 {1..1}
DISCRETE V45 {0..4}
DISCRETE V46 {3..3}
DISCRETE V47 {0..4}
DISCRETE V48 {4..4}
DISCRETE V49 {0..4}
DISCRETE V50 {0..0}
DISCRETE V51 {0..4}
DISCRETE V52 {1..1}
DISCRETE V53 {0..4}
DISCRETE V54 {2..2}
DISCRETE V55 {0..4}
DISCRETE V56 {4..4}
DISCRETE V57 {0..4}
DISCRETE V58 {0..0}
DISCRETE V59 {0..4}
DISCRETE V60 {1..1}
DISCRETE V61 {0..4}
DISCRETE V62 {2..2}
DISCRETE V63 {0..4}
DISCRETE V64 {3..3}

# aliases

**SEARCH**

PRINT [
[V0],[V1],[V2],[V3],[V4],[V5],[V6],[V7],
[V8],[V9],[V10],[V11],[V12],[V13],[V14],[V15],
[V16],[V17],[V18],[V19],[V20],[V21],[V22],[V23],
[V24],[V25],[V26],[V27],[V28],[V29],[V30],[V31],
[V32],[V33],[V34],[V35],[V36],[V37],[V38],[V39],
[V40],[V41],[V42],[V43],[V44],[V45],[V46],[V47],
[V48],[V49],[V50],[V51],[V52],[V53],[V54],[V55],
[V56],[V57],[V58],[V59],[V60],[V61],[V62],[V63],
[V64]]

VARORDER [
V0,V1,V2,V3,V4,V5,V6,V7,
V8,V9,V10,V11,V12,V13,V14,V15,
V16,V17,V18,V19,V20,V21,V22,V23,
V24,V25,V26,V27,V28,V29,V30,V31,
V32,V33,V34,V35,V36,V37,V38,V39,
V40,V41,V42,V43,V44,V45,V46,V47,
V48,V49,V50,V51,V52,V53,V54,V55,
V56,V57,V58,V59,V60,V61,V62,V63,
V64]


**CONSTRAINTS**

alldiff([ V1, V2, V3, V4])
alldiff([ V5, V10, V15, V20])
alldiff([ V5, V7, V8, V9])
alldiff([ V1, V11, V16, V21])
alldiff([ V10, V11, V13, V14])
alldiff([ V2, V7, V17, V22])
alldiff([ V15, V16, V17, V19])
alldiff([ V3, V8, V13, V23])
alldiff([ V20, V21, V22, V23])
alldiff([ V4, V9, V14, V19])
watchelement([ 0, V1, V2, V3, V4], V5, V25)
watchelement([ 0, V1, V2, V3, V4], V25, 1)
watchelement([ 0, V1, V2, V3, V4], V10, V27)
watchelement([ 0, V1, V2, V3, V4], V27, 2)
watchelement([ 0, V1, V2, V3, V4], V15, V29)
watchelement([ 0, V1, V2, V3, V4], V29, 3)
watchelement([ 0, V1, V2, V3, V4], V20, V31)
watchelement([ 0, V1, V2, V3, V4], V31, 4)
watchelement([ V5, 1, V7, V8, V9], V1, V33)
watchelement([ V5, 1, V7, V8, V9], V33, 0)
watchelement([ V5, 1, V7, V8, V9], V11, V35)
watchelement([ V5, 1, V7, V8, V9], V35, 2)
watchelement([ V5, 1, V7, V8, V9], V16, V37)
watchelement([ V5, 1, V7, V8, V9], V37, 3)
watchelement([ V5, 1, V7, V8, V9], V21, V39)
watchelement([ V5, 1, V7, V8, V9], V39, 4)
watchelement([ V10, V11, 2, V13, V14], V2, V41)
watchelement([ V10, V11, 2, V13, V14], V41, 0)
watchelement([ V10, V11, 2, V13, V14], V7, V43)
watchelement([ V10, V11, 2, V13, V14], V43, 1)
watchelement([ V10, V11, 2, V13, V14], V17, V45)
watchelement([ V10, V11, 2, V13, V14], V45, 3)
watchelement([ V10, V11, 2, V13, V14], V22, V47)
watchelement([ V10, V11, 2, V13, V14], V47, 4)
watchelement([ V15, V16, V17, 3, V19], V3, V49)
watchelement([ V15, V16, V17, 3, V19], V49, 0)
watchelement([ V15, V16, V17, 3, V19], V8, V51)
watchelement([ V15, V16, V17, 3, V19], V51, 1)
watchelement([ V15, V16, V17, 3, V19], V13, V53)
watchelement([ V15, V16, V17, 3, V19], V53, 2)
watchelement([ V15, V16, V17, 3, V19], V23, V55)
watchelement([ V15, V16, V17, 3, V19], V55, 4)
watchelement([ V20, V21, V22, V23, 4], V4, V57)
watchelement([ V20, V21, V22, V23, 4], V57, 0)
watchelement([ V20, V21, V22, V23, 4], V9, V59)
watchelement([ V20, V21, V22, V23, 4], V59, 1)
watchelement([ V20, V21, V22, V23, 4], V14, V61)
watchelement([ V20, V21, V22, V23, 4], V61, 2)
watchelement([ V20, V21, V22, V23, 4], V19, V63)
watchelement([ V20, V21, V22, V23, 4], V63, 3)
diseq(V2, 2)
diseq(V3, 3)
diseq(V7, 1)
diseq(V7, 2)
diseq(V8, 1)
diseq(V8, 3)
diseq(V9, 1)
diseq(V10, 2)
diseq(V11, 1)
diseq(V11, 2)
diseq(V13, 2)
diseq(V13, 3)
diseq(V14, 2)
diseq(V15, 3)
diseq(V16, 1)
diseq(V16, 3)
diseq(V17, 2)
diseq(V17, 3)
diseq(V21, 1)
diseq(V22, 2)

**EOF**
